OrbitQt/OverlayWidget.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_OVERLAY_WIDGET_H_
+#define ORBIT_QT_OVERLAY_WIDGET_H_
+
+#include <QEvent>
+#include <QObject>
+#include <QPainter>
+#include <QWidget>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_OverlayWidget.h"
+
+namespace OrbitQt {
+
+class OverlayWidget : public QWidget {
+  Q_OBJECT
+  Q_PROPERTY(bool spinning READ IsSpinning WRITE SetSpinning)
+  Q_PROPERTY(bool cancelable READ IsCancelable WRITE SetCancelable);
+  Q_PROPERTY(QString statusMessage READ GetStatusMessage WRITE SetStatusMessage)
+  Q_PROPERTY(QString buttonMessage READ GetButtonMessage WRITE SetButtonMessage);
+
+ public:
+  explicit OverlayWidget(QWidget* parent);
```

Comment:
It's best practive in Qt-code to have a default argument for the parent: `QWidget* parent = nullptr`

Reply:
Yes, but I did this here on purpose to indicate that this OverlayWidget needs a parent. The definition of this constructor also contains a CHECK(parent != nullptr);

Reply:
Alright. Makes sense!